File: app/repository/GameRepository.java
package repository;

import models.Game;
import play.db.ebean.EbeanConfig;

import javax.inject.Inject;
import java.util.Optional;
import java.util.concurrent.CompletionStage;


/**
 * A repository that executes database operations in a different
 * execution context.
 */
public class GameRepository extends BaseRepository<Game> {

    @Inject
    public GameRepository(EbeanConfig ebeanConfig, DatabaseExecutionContext executionContext) {
        super(ebeanConfig, executionContext);
    }

    @Override
    protected Class<Game> getModelClass() {
        return Game.class;
    }

    public CompletionStage<Optional<Game>> view(Long id) {
        return get(id, query -> query
                .fetch("tournament", "")
                .fetch("scores", "score")
                .fetch("scores.tournamentEntry.version", "name"));
    }

}

File: app/repository/GameRepository.scala
package repository

import javax.inject._
import models.Game
import play.db.ebean.EbeanConfig

import scala.concurrent.Future

class GameRepository @Inject()(val ebeanConfig: EbeanConfig, val executionContext: DatabaseExecutionContext) extends BaseRepository[Game] {
  val modelClass: Class[Game] = classOf[Game]

  def view(id: Long): Future[Option[Game]] =
    getOneWhere(id) {
      query
        .fetch("tournament", "")
        .fetch("scores", "score")
        .fetch("scores.tournamentEntry.version", "name")
    }
}
